src/functions.ts

```diff
@@ -64,26 +64,26 @@ export interface EventFunctionWithCallback {
  * A CloudEvent function handler.
  * @public
  */
-export interface CloudEventFunction {
-  (cloudEvent: CloudEvent): any;
+export interface CloudEventFunction<T = unknown> {
+  (cloudEvent: CloudEvent<T>): any;
 }
 /**
  * A CloudEvent function handler with callback.
  * @public
  */
-export interface CloudEventFunctionWithCallback {
-  (cloudEvent: CloudEvent, callback: Function): any;
+export interface CloudEventFunctionWithCallback<T = unknown> {
+  (cloudEvent: CloudEvent<T>, callback: Function): any;
 }
 /**
  * A function handler.
  * @public
  */
-export type HandlerFunction =
+export type HandlerFunction<T = unknown> =
   | HttpFunction
   | EventFunction
   | EventFunctionWithCallback
-  | CloudEventFunction
-  | CloudEventFunctionWithCallback;
+  | CloudEventFunction<T>
+  | CloudEventFunctionWithCallback<T>;
```

Comment:
Is it possible for a developer to write a CloudEventFunction without a parameterized type?

Perhaps it is not possible.

```ts
export type HandlerFunction<T = unknown> =
  | HttpFunction
  | EventFunction
  | EventFunctionWithCallback
  | CloudEventFunction<T>
  | CloudEventFunctionWithCallback<T>
  | CloudEventFunction
  | CloudEventFunctionWithCallback;
```